api/submissions/upload.go

```diff
@@ -46,6 +50,15 @@ func findChallenge(cs model.Challenges, id string) (model.Challenge, error) {
 	return cs.Find(cid)
 }
 
+func verifyUser(err error, r *http.Request, us model.Users, u **model.User) error {
```

Comment:
Feel free to adjust. Before got to that point. I declared `u model.User` then pass it to `verifyUser` as reference `&u`. However, I'm not sure why all properties default to zero values after `ss.Add(u)`, though I passed it as reference to `storeSubmission`.